Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Staging/SetPoints/SetpointController.mo

```diff
@@ -0,0 +1,596 @@
+within Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.SetPoints;
+block SetpointController
+  "Calculates the boiler stage status setpoint signal"
+
+  parameter Boolean primaryOnly = false
+    "Is the boiler plant a primary-only, condensing boiler plant?"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Integer nBoi = 2
+    "Number of boilers"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Integer boiTyp[nBoi]={
+    Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Types.BoilerTypes.condensingBoiler,
+    Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Types.BoilerTypes.nonCondensingBoiler}
+    "Boiler type"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Integer nSta = 3
+    "Number of boiler plant stages"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Integer staMat[nSta, nBoi] = {{1,0},{0,1},{1,1}}
+    "Staging matrix with stage as row index and boiler as column index"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Integer iniSta = 1
+    "Initial boiler plant stage"
+    annotation(Dialog(tab="General", group="Boiler plant configuration parameters"));
+
+  parameter Real boiDesCap[nBoi](
+    final unit="W",
+    final displayUnit="W",
```

Comment:
In the latest edits to the documentation that are now in a PR Michael would like us to not use `final` with `displayUnit`. I am planning to soon do a fix for the chiller plant package.